app/__init__.py

```diff
@@ -260,7 +260,9 @@ def exception(error):
 
     @app.errorhandler(WerkzeugHTTPException)
     def werkzeug_exception(e):
-        return make_response(jsonify(result="error", message=e.description), e.code, e.get_headers())
+        # Make sure we will send a JSON content-type by purging error provided one.
+        headers = [h for h in e.get_headers() if h[0] != "Content-Type"]
```

Comment:
The new caught exception comes with a default header for `Content-Type` that aligns to `text`. That screws up our tests as these expect to receive JSON even if that is an error. Hence I stripped the default content type that came with the exception so that the JSON content type set later can take precedence.

Reply:
does this error handler also generate a dynamic error page ? [Removing the Content-Type](https://www.rfc-editor.org/rfc/rfc7231#section-3.1.1.5) usually means browsers will assume `application/octet-stream`, or even worse trying to guess the content type based on response.

Instead of removing, do you have the option of adding `application/json` instead of dropping?

This risk only exists if this response relates to any user interactions and can be sent as a link to logged in users.

Reply:
It's not for a dynamic error page but for non-handled exceptions raised by werkzeug. The only way to have a JSON response is to actually remove the text content type from the headers provided by the exception and then passed to the `make_response` function. The call to `jsonify` on the next line will actually set a JSON content type so the response will be JSON (and be aligned to pre 2.0 version of the werkzeug library). 

To recap, the issue was that a the `headers` parameter started to have a `content-type` header which was not present before and in doing so, was overwriting the previously set content type of JSON by `jsonify`. To remove it is to let the `jsonify` function do its work.

Reply:
Thanks @jimleroyer that makes total sense!